examples/planning/add_keyword_plan.py

```diff
@@ -0,0 +1,210 @@
+
+    keyword_plan_service = client.get_service('KeywordPlanService',
+                                              version='v1')
+    response = keyword_plan_service.mutate_keyword_plans(customer_id,
+                                                         [operation])
+    resource_name = response.results[0].resource_name
+
+    print('Created keyword plan with resource name: {}'.format(resource_name))
+
+    return resource_name
+
+
+def create_keyword_plan_campaign(client, customer_id, keyword_plan):
+    operation = client.get_type('KeywordPlanCampaignOperation', version='v1')
+    keyword_plan_campaign = operation.create
+
+    keyword_plan_campaign.name.value = 'Keyword plan campaign {}'.format(
+        uuid.uuid4())
+    keyword_plan_campaign.cpc_bid_micros.value = 1000000
+    keyword_plan_campaign.keyword_plan.value = keyword_plan
+
+    keyword_plan_network = client.get_type('KeywordPlanNetworkEnum',
+                                           version='v1')
+    network = keyword_plan_network.GOOGLE_SEARCH
+    keyword_plan_campaign.keyword_plan_network = network
+
+    geo_target = client.get_type('KeywordPlanGeoTarget', version='v1')
+    # Constant for U.S. Other geo target constants can be referenced here:
+    # https://developers.google.com/adwords/api/docs/appendix/geotargeting
+    geo_target.geo_target_constant.value = 'geoTargetConstants/2840'
+    keyword_plan_campaign.geo_targets.extend([geo_target])
+
+    language = client.get_type('StringValue', version='v1')
+    # Constant for English
+    language.value = 'languageConstants/1000'
+    keyword_plan_campaign.language_constants.extend([language])
+
+    keyword_plan_campaign_service = client.get_service(
+        'KeywordPlanCampaignService', version='v1')
+    response = keyword_plan_campaign_service.mutate_keyword_plan_campaigns(
+        customer_id, [operation])
+
+    resource_name = response.results[0].resource_name
+
+    print('Created keyword plan campaign with resource name: {}'.format(
+        resource_name))
+
+    return resource_name
+
+
+def create_keyword_plan_ad_group(client, customer_id, keyword_plan_campaign):
+    operation = client.get_type('KeywordPlanAdGroupOperation', version='v1')
+    keyword_plan_ad_group = operation.create
+
+    keyword_plan_ad_group.name.value = 'Keyword plan ad group {}'.format(
+        uuid.uuid4())
+    keyword_plan_ad_group.cpc_bid_micros.value = 2500000
+    keyword_plan_ad_group.keyword_plan_campaign.value = keyword_plan_campaign
+
+    keyword_plan_ad_group_service = client.get_service(
+        'KeywordPlanAdGroupService', version='v1')
+    response = keyword_plan_ad_group_service.mutate_keyword_plan_ad_groups(
+        customer_id, [operation])
+
+    resource_name = response.results[0].resource_name
+
+    print('Created keyword plan ad group with resource name: {}'.format(
+        resource_name))
+
+    return resource_name
+
+
+def create_keyword_plan_keywords(client, customer_id, plan_ad_group):
```

Comment:
Could you please add docs for these methods so it is easy to follow?

Reply:
Yes great idea.